boa/contracts/base_evm_contract.py

```diff
@@ -37,7 +37,8 @@ def handle_error(self, computation):
 
     @property
     def address(self) -> Address:
-        assert self._address is not None
+        if self._address is None:
+            raise ValueError("Contract address is not set")
```

Comment:
```suggestion
            # avoid assert, in pytest it would call repr(self) which recurses again
            raise RuntimeError("Contract address is not set")
```